profiling/energy-profiler/daemon.hpp

```diff
@@ -0,0 +1,39 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#pragma once
+
+#include <functional>
+#include <thread>
+#include <chrono>
+
+class Daemon {
+ public:
+  Daemon(std::function<void()> func, int interval_ms)
```

Comment:
Should we pass in a `std::chrono::duration` directly, instead of an `int`? What about the type `int`; e.g. what is the default arithmetic type that `std::chrono::milliseconds` encodes?

Reply:
yes, that is a good idea.